### DISCUSSION TITLE
Renovate python package manager support different versions of python

### DISCUSSION THREAD

**Discussion by alaamub** [renovatebot/renovate#13001] [Request Help]
So we have few projects that only support python 2.7 and few others support 3.5. when updating dependencies, does renovate has an understanding of bumping version that aligns with python version. 

**Example**
if `python==3.7.3`	  then numpy==1.21.4 is supported.
if `python==2.7.16`   then numpy==1.16.6 is supported.

How can we configure renovate so it knows what the maximum version to update to that aligns with python version.

**Comment by rarkins** [comment]
There is some basic support for this concept already, although I can't be sure it satisfies your use case. A reproduction repo if it doesn't work would be best.

**Reply by alaamub** [reply]
thanks @rarkins can you please point to the basic support configuration for this, I can try and get back to you.

**Reply by rarkins** [reply]
https://github.com/renovatebot/renovate/blob/705b6db57bf09da80040b0f3cb1000db04f479bd/lib/datasource/index.ts#L347-L367

**Reply by alaamub** [reply]
oh awesome, thanks @rarkins

**Reply by alaamub** [reply]
Update: I used constraints and it's working, thanks!

I'm curios, since I have multiple projects within the same monorepo that contain multiple python version support e.g 2.7, 3.5 and 3.6, looks like `force` and `constraints`  can not be applied within the packageRules. is it possible to create it within package rules?

```
 "force": {
    	"constraints": {
          "python": "2.7.11"
	}
    },
```

**Reply by alaamub** [reply]
Hi @rarkins kindly any update on this?

**Reply by rarkins** [reply]
Force cannot be used

**Reply by alaamub** [reply]
Got it thanks!
